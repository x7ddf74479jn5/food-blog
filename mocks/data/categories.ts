export const categories = {
  rice: { id: "1", slug: "rice", name: "ご飯物", color: "grey" },
  salad: { id: "2", slug: "salad", name: "サラダ", color: "grey" },
  tips: { id: "3", slug: "tips", name: "料理基礎", color: "grey" },
};
