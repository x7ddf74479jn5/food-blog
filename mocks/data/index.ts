export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
