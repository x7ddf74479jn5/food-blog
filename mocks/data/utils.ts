export const dateCommon = {
  createdAt: "2020-12-12T00:00:00+09:00",
  updatedAt: "2020-12-12T00:00:00+09:00",
  publishedAt: "2020-12-12T00:00:00+09:00",
  revisedAt: "2020-12-12T00:00:00+09:00",
};
