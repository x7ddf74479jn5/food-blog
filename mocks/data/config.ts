import type { TConfig } from "@/types/models/config";

import { dateCommon } from "./utils";

export const mockConfig: TConfig = {
  ...dateCommon,
  apiHost: "https://food-blog.microcms.io/api/v1/",
  host: "https://food-blog-chi.vercel.app/",
  siteTitle: "Food Blog",
  siteDescription: "料理レシピを紹介するブログ",
  siteKeywords: ["料理"],
  siteImage: { url: "https://food-blog-chi.vercel.app/site_image.jpg", width: 100, height: 100 },
  organization: "Pandashark",
  officialSite: "https://next-portfolio-livid.vercel.app/",
  twitterId: "@pandashark6",
  email: "x7ddf74479jn5@gmail.com",
};
