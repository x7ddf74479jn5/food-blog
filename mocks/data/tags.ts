import type { TTag } from "@/types";

import { dateCommon } from "./utils";

type TTagCollection = {
  [key: string]: TTag;
};

export const mockTags: TTagCollection = {
  komatsuna: { ...dateCommon, id: "1", slug: "komatshna", name: "小松菜", color: "green" as const },
  mozzarella: { ...dateCommon, id: "2", slug: "mozzarella", name: "モッツァレラ", color: "pink" as const },
  lettuce: { ...dateCommon, id: "3", slug: "lettuce", name: "レタス", color: "green" as const },
  mincedMeat: { ...dateCommon, id: "4", slug: "minced_meat", name: "挽き肉", color: "red" as const },
  rice: { ...dateCommon, id: "5", slug: "rice", name: "ごはん", color: "yellow" as const },
  aburaage: { ...dateCommon, id: "6", slug: "aburaage", name: "油揚げ", color: "gray" as const },
  onion: { ...dateCommon, id: "7", slug: "onion", name: "玉ねぎ", color: "green" as const },
  udon: { ...dateCommon, id: "8", slug: "udon", name: "うどん", color: "yellow" as const },
  misoSoup: { ...dateCommon, id: "9", slug: "miso_soup", name: "味噌汁", color: "gray" as const },
  tomato: { ...dateCommon, id: "10", slug: "tomato", name: "トマト", color: "green" as const },
  preparation: { ...dateCommon, id: "11", slug: "preparation", name: "仕込み", color: "gray" as const },
  ooba: { ...dateCommon, id: "12", slug: "ooba", name: "大葉", color: "green" as const },
};
