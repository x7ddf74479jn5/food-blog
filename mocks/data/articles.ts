import type { TArticle } from "@/types";

import { categories } from "./categories";
import { tags } from "./tags";
import { dateCommon } from "./utils";
import { writers } from "./writers";

const articleStock = {
  id: "1",
  title: "基本の一番だしの作り方",
  writer: writers.pandashark,
  description:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  image: { url: "/images/5026416_s.jpg", height: 427, width: 640 },
  excerpt:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  body: `**分量： 800ml**

だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。

## 材料
- 水：1000ml
- だし昆布：10g
- かつお削り節：40g

## 作り方
- 固く絞った布巾で、昆布の表面の汚れを軽く拭き取ります。
- 鍋に水と昆布を入れます。
- 弱火で７分ほど熱し、鍋底から小さい泡が出てきたら昆布を取り出します。
- 昆布だしを一旦沸騰させてから、削り節を入れ、弱火で１分煮出します。
- ボウルの上にザルを置き、その上にキッチンペーパーを敷いて出汁を濾します。

---

**二番だしの取り方**
1. 鍋に 500ml の水を入れて沸騰させます。
2. 鍋に一番だしのだしがらを入れ、３分煮出します。
3. 一番だしと同じ方法で濾します。

---

## 保存方法
使わない分は保存容器に移して、冷蔵庫で保存してください。冷蔵保存で 2〜3 日は持ちます。

##  ポイント
だし昆布の表面の白い粉は「うま味」成分なので、取り除かずにそのままお使いください。`,
  category: categories.rice,
  tags: [tags.preparation, tags.misoSoup, tags.udon],
  imageOption: null,
};

type TArticleCollection = {
  [key: string]: TArticle;
};

export const articles: TArticleCollection = {
  stock: { ...dateCommon, ...articleStock },
};
