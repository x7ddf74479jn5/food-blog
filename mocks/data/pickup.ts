import { mockArticles } from "@mocks/data";
import { dateCommon } from "@mocks/data/utils";

export const mockPickup = {
  id: "1",
  name: "旬の食材を使ったおすすめレシピ",
  description: "旬の食材を使ったおすすめレシピ",
  startDate: new Date("2021/10/08 00:00"),
  endDate: new Date("2023/02/08 00:00"),
  articles: Object.values(mockArticles),
  ...dateCommon,
};
