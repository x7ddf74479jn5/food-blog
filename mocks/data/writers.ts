import { dateCommon } from "@mocks/data/utils";
import type { MicroCMSListContent } from "microcms-js-sdk";

import type { TWriter } from "@/types";

type TMockWriterKey = "pandashark";

type TWriterCollection = {
  [P in TMockWriterKey]: TWriter & MicroCMSListContent;
};

export const mockWriters: TWriterCollection = {
  pandashark: {
    ...dateCommon,
    id: "1",
    name: "pandashark",
    fullName: "pandashark",
    description: "description",
    twitterAccountName: "@pandashark",
    avatar: { url: "pandashark_logo_180x180.png", width: 100, height: 100 },
  },
};
