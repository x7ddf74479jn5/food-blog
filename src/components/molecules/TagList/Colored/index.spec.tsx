import { mockTags } from "@mocks/data";
import React from "react";
import renderer from "react-test-renderer";

import { TagListColored } from ".";

describe("components/molecules/TagListColored", () => {
  const mockTagList = Object.values(mockTags);

  it("snapshot", () => {
    const tree = renderer.create(<TagListColored tags={mockTagList} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
