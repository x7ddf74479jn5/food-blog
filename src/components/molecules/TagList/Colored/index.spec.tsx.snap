// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/TagListColored snapshot 1`] = `
<ul
  className="flex flex-row flex-wrap gap-2 items-center"
>
  <li>
    <a
      href="/articles/tags/komatshna"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-green-700 rounded-lg"
      >
        #
        小松菜
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/mozzarella"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-pink-700 rounded-lg"
      >
        #
        モッツァレラ
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/lettuce"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-green-700 rounded-lg"
      >
        #
        レタス
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/minced_meat"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-red-700 rounded-lg"
      >
        #
        挽き肉
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/rice"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-yellow-700 rounded-lg"
      >
        #
        ごはん
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/aburaage"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-gray-500 rounded-lg"
      >
        #
        油揚げ
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/onion"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-green-700 rounded-lg"
      >
        #
        玉ねぎ
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/udon"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-yellow-700 rounded-lg"
      >
        #
        うどん
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/miso_soup"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-gray-500 rounded-lg"
      >
        #
        味噌汁
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/tomato"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-green-700 rounded-lg"
      >
        #
        トマト
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/preparation"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-gray-500 rounded-lg"
      >
        #
        仕込み
      </div>
    </a>
  </li>
  <li>
    <a
      href="/articles/tags/ooba"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="inline-block py-1 px-2 text-white  bg-green-700 rounded-lg"
      >
        #
        大葉
      </div>
    </a>
  </li>
</ul>
`;
